test/conftest.py

```diff
@@ -205,3 +188,127 @@ def mc():
 @pytest.fixture(scope="function")
 def db_connection() -> psycopg2.extensions.connection:
     return psycopg2.connect(DB_CONNINFO)
+
+
+def name_db_schema_main(project_name: str) -> str:
+    return project_name + "_main"
+
+
+def name_db_schema_base(project_name: str) -> str:
+    return project_name + "_base"
+
+
+def name_project_dir(project_name: str) -> str:
+    return os.path.join(
+        TMP_DIR,
+        project_name + "_work",
+    )
+
+
+def name_project_sync_dir(project_name: str) -> str:
+    return os.path.join(
+        TMP_DIR,
+        project_name + "_dbsync",
+    )
+
+
+def complete_project_name(project_name: str) -> str:
+    return WORKSPACE + "/" + project_name
+
+
+def path_test_data(filename: str) -> str:
+    return os.path.join(
+        TEST_DATA_DIR,
+        filename,
+    )
+
+
+def filename_sync_gpkg() -> str:
+    return "test_sync.gpkg"
+
+
+def init_sync_from_db(mc: MerginClient, project_name: str, ignored_tables: List[str] = None):
+    """
+    Initialize sync from given database file:
+    - prepare schema with simple table
+    - create MM project
+    - configure DB sync and let it do the init
+    """
+    if ignored_tables is None:
+        ignored_tables = []
+
+    full_project_name = complete_project_name(project_name)
+    project_dir = name_project_dir(project_name)  # working directory
+    sync_project_dir = name_project_sync_dir(project_name)  # used by dbsync
+    db_schema_main = name_db_schema_main(project_name)
+    db_schema_base = name_db_schema_base(project_name)
+
+    conn = psycopg2.connect(DB_CONNINFO)
+
+    cleanup(
+        mc,
+        full_project_name,
+        [
+            project_dir,
+            sync_project_dir,
+        ],
+    )
+    cleanup_db(
+        conn,
+        db_schema_base,
+        db_schema_main,
+    )
+
+    with open(
+        path_test_data("create_base.sql"),
+        encoding="utf-8",
+    ) as file:
+        base_table_dump = file.read()
+
+    base_table_dump = base_table_dump.replace("default-schema-name", db_schema_main)
+    base_table_dump = base_table_dump.replace("default-table-name", "simple")
```

Comment:
I would prefer not to do this kind of replacement, and simply have .sql files with the expected schema/table names hardcoded. I think we do not need to have a different schema for each test - in theory all the tests could use the same "main" schema name in DB, they are run sequentially so it should not hurt, and it would simplify things (e.g. one could also copy&paste the commands from .sql file to a sql console without modifications)